pkg/ciliumenvoyconfig/cec_manager.go

```diff
@@ -325,6 +334,16 @@ func (r *cecManager) updateCiliumEnvoyConfig(
 	oldCECObjectMeta metav1.ObjectMeta, oldCECSpec *ciliumv2.CiliumEnvoyConfigSpec,
 	newCECObjectMeta metav1.ObjectMeta, newCECSpec *ciliumv2.CiliumEnvoyConfigSpec,
 ) error {
+	if oldCECObjectMeta.GetNamespace() == "" {
```

Comment:
couldn't we get rid of this completely? as decreasing and immediately increasing the metric is a "no-op" (as the underlying metric isn't interested in the spec)